### PR TITLE
reduce IdModel graph build in AllocationOrderInference

### PR DESCRIPTION
**Pull request by jjsjann123**

### CODE REVIEW

File: csrc/id_model/id_model.cpp [hunk 1]
```diff
@@ -1263,6 +1263,7 @@ Expr* IdModel::addReplayAs(std::vector<IterDomain*> new_inputs, Expr* expr) {
 
   // Create the new expression with provided inputs
   auto replay = ReplayTransform::replayAs(new_inputs, expr);
+  NVF_ERROR(replay != nullptr, "no replay found");
```
Comment (jjsjann123) [inline csrc/id_model/id_model.cpp:1266]: cc'ing @naoyam 

Does `buildLoopGraph` have some assumption on what a fusion graph should look like, are we not supposed to build all graph from an unscheduled fusion with IdModel? I can open an issue if that's not the case. This PR is just a quick patch to make our thunder benchmark green again.

**Comment by jjsjann123** [comment]
!build

**Comment by naoyam** [comment]
Can you please give some info about what the issue is?

**Comment by jjsjann123** [comment]
> Can you please give some info about what the issue is?

I have a issue opened on this: https://github.com/NVIDIA/Fuser/issues/2253

File: csrc/id_model/id_model.cpp (hunk 1 above)
Comment (jjsjann123) [reply csrc/id_model/id_model.cpp:1266]: tagging #2253

**Review by naoyam — APPROVED** [review]
LGTM

**Pull request merged**